server/jetstream_api.go

```diff
@@ -1480,7 +1502,17 @@ func (s *Server) jsStreamNamesRequest(sub *subscription, c *client, subject, rep
 	resp.Total = numStreams
 	resp.Limit = JSApiNamesLimit
 	resp.Offset = offset
-	s.sendAPIResponse(ci, acc, subject, reply, string(msg), s.jsonResponse(resp))
+	if responseDelay != 0 {
+		// Need to copy these off before sending..
+		msg = append(msg[:0:0], msg...)
```

Comment:
For info if later this PR is accepted or revisited: Since sendAPIResponse() seem to accept a string for the "msg" and not a slice, (see below `string(msg)`) you should capture the conversion as a string here, before starting the go routine. Something like this:
```
strMsg := string(msg)
s.startGoRoutine(func() {
			defer s.grWG.Done()
			time.Sleep(responseDelay)
			s.sendAPIResponse(ci, acc, subject, reply, strMsg, s.jsonResponse(resp))
		})
```
This applies to all places where you made a copy because of sending in a separate go routine.